safe-global/web-core
# feat: AB testing system

## What it solves

Partially resolves #968

## How this PR fixes it

A new AB testing system has been implemented, adding `abTest` to GTM events (if `_abTest` is currently set).

## How to test it

`gtmSetAbTest` should be called in `useEffect` and the returned unsubscribe function called on cleanup. When set, the `abTest` key should exist in the GTM events and be visible on debug view.

## Analytics changes

`abTest` string should now be present in events.

## Screenshots

![image](https://user-images.githubusercontent.com/20442784/199016659-7a6812e2-6607-4226-be40-500635dd7ca1.png)
